src/components/Planner/Planner.tsx

```diff
@@ -855,31 +859,34 @@ const Planner = () => {
-                                                          top: "-3.5px",
-                                                          right: "-3.5px",
-                                                          opacity: isHovering
-                                                            ? 1
-                                                            : 0,
-                                                          transition:
-                                                            "opacity 0.3s ease-in-out"
-                                                        }}
-                                                        mr="0rem"
-                                                        size="xs"
-                                                        onClick={() =>
-                                                          handleDelete(
-                                                            item.mealId
-                                                          )
-                                                        }
-                                                      />
-                                                    )}
+                                                    {item.mealId &&
+                                                      isHovering[
+                                                        item.mealId
+                                                      ] && (
+                                                        <IconButton
+                                                          aria-label="Delete recipe"
+                                                          icon={<CloseIcon />}
+                                                          style={{
+                                                            position:
+                                                              "absolute",
+                                                            top: "-3.5px",
+                                                            right: "-3.5px",
+                                                            opacity: isHovering
+                                                              ? 1
+                                                              : 0,
+                                                            transition:
+                                                              "opacity 0.3s ease-in-out"
+                                                          }}
+                                                          mr="0rem"
+                                                          size="xs"
+                                                          onClick={() =>
+                                                            handleDelete(
+                                                              // Checked that this exists before rendering this component
+                                                              item.mealId!
```

Comment:
This one is my code editor not interpreting things properly. The `!` explicitly tells typescript that I _know_ this exists (because I checked for it on line 862).